dali/python/nvidia/dali/pipeline.py

```diff
@@ -1203,6 +1244,22 @@ def iter_setup(self):
         data from NumPy arrays."""
         pass
 
+    def _generate_build_args(self, output_dtype, output_ndim):
+        ret = []
+        num_outputs = len(self._names_and_devices)
+        output_dtype = [output_dtype] * num_outputs if type(
+            output_dtype) is not list else output_dtype
+        output_ndim = [output_ndim] * num_outputs if type(output_ndim) is not list else output_ndim
+        if not (len(output_dtype) == len(output_ndim) == num_outputs):
+            raise RuntimeError(
+                f"Inconsistent output description. Length of provided output descriptions do not match. \n"
+                f"Expected num_outputs={num_outputs}.\nReceived:\noutput_dtype={output_dtype}\noutput_ndim={output_ndim}")
+
+        for nd, dtype, ndim in zip(self._names_and_devices, output_dtype, output_ndim):
```

Comment:
What is names_and_devices? Is it tuple? Would this work:
```suggestion
        for (name, device), dtype, ndim in zip(self._names_and_devices, output_dtype, output_ndim):
```

Reply:
Yes it would